feature_engine/feature_selection.py

```diff
@@ -88,3 +89,138 @@ def transform(self, X):
+
+    Attributes
+    ----------
+
+    constant_features_: list
+        List of constant features in a dataframe
+
+    quasi_constant_features_: list
+        List of quasi constant features in a dataframe
+
+    selected_features_: list
+        List of selected features which are non-constant and non-quasi constant
+
+    """
+    def __init__(self, drop_quasi_constant_features=False, quasi_constant_threshold=None):
+
+        self.drop_quasi_constant_features = drop_quasi_constant_features
+        self.quasi_constant_threshold = quasi_constant_threshold
+
+        self.constant_features_ = []
+        self.selected_features_ = []
+        self.quasi_constant_features_ = []
+        self._input_shape_ = ()
+
+        if not self.drop_quasi_constant_features and self.quasi_constant_threshold is not None:
+            warnings.warn(
+                'Setting a quasi_constant_threshold has no effect since drop_quasi_constant_features is False. You '
+                'should leave quasi_constant_threshold to its default (None), or set drop_quasi_constant_features=True.'
+                , FutureWarning
+            )
+
+        if self.drop_quasi_constant_features:
+            if self.quasi_constant_threshold <= 0 or self.quasi_constant_threshold >= 1:
+                raise ValueError("Value of threshold should be between 0 and 1")
+
+    def fit(self, X, y=None):
+
+        """
+        Find constant and non-quasi constant features from a dataframe
+
+        Parameters
+        ----------
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe
+
+        y: None
+            y is not needed for this transformer. You can pass y or None.
+
+        """
+
+        # data frame check
+        X = _is_dataframe(X)
+
+        # get input shape
+        self._input_shape_ = X.shape
+
+        # constant features
+        self.constant_features_ = [feat for feat in X.columns if X[feat].nunique() == 1]
+
+        # non-constant features
+        self.selected_features_ = [feat for feat in X.columns if feat not in self.constant_features_]
+
+        # if total constant features is equal to total features raise an error
+        if len(self.constant_features_) == len(X.columns):
+            raise ValueError("The resulting dataframe will have no columns after dropping all constant features")
+
+        if self.drop_quasi_constant_features:
+            # quasi-constant features
+            self.quasi_constant_features_ = [feat for feat in X.columns if X[feat].value_counts(
+                normalize=True, sort=True, ascending=False).iloc[0] > self.quasi_constant_threshold
+                                             and feat not in self.constant_features_
+                                             and feat in self.selected_features_]
+
+            self.selected_features_ = [feat for feat in self.selected_features_
+                                       if feat not in self.quasi_constant_features_]
+
```

Comment:
this is good. I wondered if we can optimize the code by using sets and check for differences taking advantage of python functionality? instead of the list comprehension?

Reply:
I did not intentionally use set because it produces unordered elements and it would disrupt the column ordering in original df and could cause inconvenience. Like in below df, I remove D and set produces [B,A,C,E] while with LC ordering is preserved which is [A,B,C,E]. I could go with set just wanted to avoid the issue here. Let me know your thoughts.  

<img width="778" alt="set" src="https://user-images.githubusercontent.com/22824809/93661434-9c79c900-fa0c-11ea-8516-3c87c9b928cb.png">

Reply:
ok, makes sense. Thanks for the explanation. I don't think we would need this bit of code though, when we refurbish the transformer as per the comments in the review.